indico/modules/events/registration/controllers/display.py

```diff
@@ -104,37 +105,74 @@ class RHParticipantList(RHRegistrationFormDisplayBase):
+
+            column_names = registration_settings.get(self.event, 'participant_list_columns')
+            headers = [PersonalDataType[column_name].get_title() for column_name in column_names]
+
+            query = (Registration
+                     .find(Registration.event_id == self.event.id,
+                           Registration.state == RegistrationState.complete,
+                           RegistrationForm.publish_registrations_enabled,
+                           ~RegistrationForm.is_deleted,
+                           ~Registration.is_deleted,
+                           _join=Registration.registration_form,
+                           _eager=Registration.registration_form)
+                     .order_by(*Registration.order_by_name))
+            registrations = [_process_registration(reg, column_names) for reg in query]
+            table = {'headers': headers, 'rows': registrations}
+            table['show_checkin'] = any(registration['checked_in'] for registration in registrations)
+            return table
+
+        def _participant_list_table(regform):
+            def _process_registration(reg, column_ids):
+                columns = [{'text': reg.data_by_field[column_id].friendly_data} for column_id in column_ids]
+                return {'checked_in': _is_checkin_visible(reg), 'columns': columns}
+            active_fields = {field.id: field.title for field in regform.active_fields}
+            column_ids = [column_id
+                          for column_id in registration_settings.get_participant_list_columns(self.event, regform)
+                          if column_id in active_fields]
+            headers = [active_fields[column_id].title() for column_id in column_ids]
+            registrations = [_process_registration(reg, column_ids) for reg in regform.active_registrations]
+            table = {'headers': headers, 'rows': registrations, 'title': regform.title}
+            table['show_checkin'] = any(registration['checked_in'] for registration in registrations)
+            return table
+
+        if registration_settings.get(self.event, 'merge_registration_forms'):
+            tables = [_merged_participant_list_table()]
+        else:
+            tables = []
+            regforms_dict = {regform.id: regform for regform in regforms if regform.publish_registrations_enabled}
+            for form_id in registration_settings.get_participant_list_form_ids(self.event):
+                try:
+                    tables.append(_participant_list_table(regforms_dict[form_id]))
+                except KeyError:
+                    # The settings might reference forms that are not available anymore (publishing was disabled, etc.)
+                    pass
+                del regforms_dict[form_id]
+            # There might be forms that have not been sorted by the user yet
+            for form_id, regform in regforms_dict.iteritems():
```

Comment:
`tables.update(map(_participant_list_table, regforms_dict.viewvalues())` would be an alternative to the loop